Docs: Update canvas docs for 9.4 (#62923)
grafana/grafana: docs/sources/panels-visualizations/visualizations/canvas/index.md

The icon element enables you to add a supported icon to the canvas. Icons can have their color set based on thresholds / value mappings.

## Connections

> When building a canvas panel, you can connect elements together to create more complex visualizations. Connections are created by dragging from the connection anchor of one element to the connection anchor of another element. Connections can also be created to the background of the canvas panel. Connection anchors are displayed when an element is hovered over and inline editing is turned on. To remove a connection, simply click on the connection directly and then press the "Delete" key.

Reviewer: Removing some of the passive voice

```suggestion
When building a canvas panel, you can connect elements together to create more complex visualizations. Connections are created by dragging from the connection anchor of one element to the connection anchor of another element. You can also create connections to the background of the canvas panel. Connection anchors are displayed when you hover over an element and inline editing is turned on. To remove a connection, simply click on the connection directly and then press the "Delete" key.
```

{{< video-embed src="/static/img/docs/canvas-panel/canvas-connections-9-4-0.mp4" max-width="750px" caption="Canvas connections demo" >}}

> ## Canvas Editing

Reviewer: ```suggestion
## Canvas editing
```

The icon element enables you to add a supported icon to the canvas. Icons can have their color set based on thresholds / value mappings.

### Server

The server element enables you to easily represent a single server, a stack of servers, a database, or a terminal. Server elements support status color, bulb color, and a bulb blink rate all configurable by fixed or field values.

{{< figure src="/media/docs/grafana/canvas-server-element-9-4-0.png" max-width="750px" caption="Canvas server element" >}}

## Connections

> When building a canvas panel, you can connect elements together to create more complex visualizations. Connections are created by dragging from the connection anchor of one element to the connection anchor of another element. You can also create connections to the background of the canvas panel. Connection anchors are displayed when you hover over an element and inline editing is turned on. To remove a connection, simply click on the connection directly and then press the "Delete" key.

Reviewer 2: Delete key or backspace, right?

Author: yes - do you think it is worth mentioning both?

Reviewer 2: Do we have a tooltip when a connection is highlighted that explicitly provides these instructions? If not, I think it is worth mentioning both in the docs for completeness.

{{< figure src="/static/img/docs/canvas-panel/canvas-context-menu-9-2-0.png" max-width="750px" caption="Canvas element context menu" >}}

## Canvas options

### Inline editing

The inline editing toggle enables you to lock or unlock the canvas panel. When turned off the canvas panel becomes “locked”, freezing elements in place and preventing unintended modifications.

{{< video-embed src="/static/img/docs/canvas-panel/canvas-inline-editing-toggle-9-2-0.mp4" max-width="750px" caption="Inline editing toggle demo" >}}

### Data links

Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you've added a data link to the panel, you can display it by following these steps:

1. Set an element to be tied to a field value.

> 1. Turn off the inline editing toggle.

Reviewer 2: Are these numbers supposed to increment?

Author: I believe they auto increment - [see this comment](https://github.com/grafana/grafana/pull/62923#discussion_r1100398712)

Reviewer 2: Nice!

1. Hover over the element to display the data link tooltip.
1. Click on the element to be able to open the data link.

> {{< video-embed src="/media/docs/grafana/canvas-data-links-9-4-0.mp4" max-width="750px" caption="Data links demo" >}}

Reviewer 2: Are embedded videos supported in docs now? If so, cool!

The inline editing toggle enables you to lock or unlock the canvas panel. When turned off the canvas panel becomes “locked”, freezing elements in place and preventing unintended modifications.

{{< video-embed src="/static/img/docs/canvas-panel/canvas-inline-editing-toggle-9-2-0.mp4" max-width="750px" caption="Inline editing toggle demo" >}}

> ### Data Links

Reviewer: ```suggestion
### Data links
```

> Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you have added a data link to the panel, you can display it following these steps:

Reviewer: ```suggestion
Canvas supports [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). Once you've added a data link to the panel, you can display it by following these steps:
```

> 1. Set an element to be tied to a field value.
> 2. Turn off the inline editing toggle.
> 3. Hover over the element to display the data link tooltip.
> 4. Click the data link tooltip to open the data link.

Reviewer: ```suggestion
1. Set an element to be tied to a field value.
1. Turn off the inline editing toggle.
1. Hover over the element to display the data link tooltip.
1. Click the data link tooltip to open the data link.
```

> 1. Set an element to be tied to a field value
> 2. Turn off the inline editing toggle
> 3. Now hover over the element and a data link tooltip will appear
> 4. Click the data link tooltip to open the data link

Reviewer: Adding missing punctuation and a few wording edits

```suggestion
1. Set an element to be tied to a field value.
2. Turn off the inline editing toggle.
3. Hover over the element to display the data link tooltip.
4. Click the data link tooltip to open the data link.
```
